pyomo/solvers/tests/checks/test_cplex.py

```diff
@@ -54,5 +61,195 @@ def test_validate_file_name(self):
+        m.c2 = Constraint(expr=quicksum(m.y[i] for i in m.s) >= 10)
+
+        m.priority = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+        m.direction = Suffix(direction=Suffix.EXPORT, datatype=Suffix.INT)
+
+        m.priority.set_value(m.x, 1)
+
+        # Ensure tests work for both options of `expand`
+        m.priority.set_value(m.y, 2, expand=False)
+        m.direction.set_value(m.y, -1, expand=True)
+
+        m.direction.set_value(m.y[10], 1)
+        return m
+
+    def test_use_variable_priorities(self):
+        model = self.get_mock_model_with_priorities()
+        with SolverFactory('cplex', solver_io='lp') as opt:
+            opt.solve(model, priorities=True, keepfiles=True)
+
+            with open(opt._priorities_file_name, 'r') as ord_file:
+                priorities_file = ord_file.read()
+
+        assert priorities_file == (
+            '* ENCODING=ISO-8859-1\n'
+            'NAME             Priority Order\n'
+            '  x1 1\n'
+            ' DN x2 2\n'
+            ' DN x3 2\n'
+            ' DN x4 2\n'
+            ' DN x5 2\n'
+            ' DN x6 2\n'
+            ' DN x7 2\n'
+            ' DN x8 2\n'
+            ' DN x9 2\n'
+            ' DN x10 2\n'
+            ' UP x11 2\n'
+            'ENDATA\n'
+        )
+
+        assert "read %s\n" % (opt._priorities_file_name,) in opt._command.script
+
+    def test_ignore_variable_priorities(self):
+        model = self.get_mock_model_with_priorities()
+        with SolverFactory('cplex', solver_io='lp') as opt:
+            opt.solve(model, priorities=False, keepfiles=True)
+
+            assert opt._priorities_file_name is None
+            assert ".ord" not in opt._command.script
+
+    def test_can_use_manual_priorities_file_with_lp_solve(self):
+        """ Test that we can pass an LP file (not a pyomo model) along with a priorities file to `.solve()` """
+        model = self.get_mock_model_with_priorities()
+
+        with SolverFactory('cplex', solver_io='lp') as pre_opt:
+            pre_opt._presolve(model, priorities=True, keepfiles=True)
+            lp_file = pre_opt._problem_files[0]
+            priorities_file_name = pre_opt._priorities_file_name
+
+        with SolverFactory('cplex', solver_io='lp') as opt:
+            opt.solve(lp_file, priorities=True, priorities_file=priorities_file_name, keepfiles=True)
+
+            assert ".ord" in opt._command.script
+
+            with open(opt._priorities_file_name, 'r') as ord_file:
+                priorities_file = ord_file.read()
+
+        assert priorities_file == (
+            '* ENCODING=ISO-8859-1\n'
+            'NAME             Priority Order\n'
+            '  x1 1\n'
+            ' DN x2 2\n'
+            ' DN x3 2\n'
+            ' DN x4 2\n'
+            ' DN x5 2\n'
+            ' DN x6 2\n'
+            ' DN x7 2\n'
+            ' DN x8 2\n'
+            ' DN x9 2\n'
+            ' DN x10 2\n'
+            ' UP x11 2\n'
+            'ENDATA\n'
+        )
```

Comment:
Perhaps this assertion is neater if you read the contents of `priorities_file_name` before doing the final solve and comparing that against `priorities_file`? They should be the same right?

Reply:
Sure